definitions/artifacts/azure-service-principal.json

```diff
@@ -245,9 +245,9 @@
         "client_secret": {
           "type": "string",
           "title": "Client Secret",
-          "pattern": "^[a-zA-Z0-9~._-]+$",
+          "pattern": "^((?![0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}).)*$",
           "message": {
-            "pattern": "Must be a valid Azure App Client Secret"
+            "pattern": "Not a valid Client Secret. You have likely added the client secret key instead of the client secret value."
```

Comment:
```suggestion
            "pattern": "Not a valid Client Secret. You have likely added the client secret ID instead of the client secret value."
```

Reply:
Shows up as ID in Azure portal